### ISSUE TITLE
How to navigate from ShareExtension to MainApp 

### ISSUE THREAD

**Issue by BigPun86** [alinz/react-native-share-extension#86]
Maybe i am not correctly understanding how the Share Extension should work so someone could correct me if necessary :) 

I actually want to use the ShareExtension in order to Share an Image with my App. I set up everything correctly and the share modal from the Sample shows up. Now how do i actually tell the ShareExtension to enter the MainApp with its value & type?

Best regards

**Comment by aliszafar1** [comment]
+1

**Comment by isaachinman** [comment]
I am also wondering the same thing. In general, I wonder what the reason is for splitting the ShareExtension into a separate application, as in most use cases you're going to want to handle the received data in your main app. Maybe someone can explain this to me.

Aside from that, it seems to me the way to go is to have your ShareExtension do some simple data transformation and then launch your main app via `Linking.openURL`, with the appropriate data as params. Obviously you'll need to have deep linking set up in your app for this to work.

Currently I am looking into how to even enable the linking API inside this ShareExtension. I'll update this comment as I progress. Good luck!

**Comment by isaachinman** [comment]
@BigPun86 @aliszafar1 Please check out the discussion on #51. There's a PR on this repository that is incredibly simple and allows you to open your host app via your share extension.

Then, just do something like this in your share receiver component, depending on your individual needs:

  async componentDidMount() {
    try {
      const data = await ShareExtension.data()
      const urlSafeData = JSON.stringify({ data })
      await ShareExtension.openURL(`myapp://ShareReceive?shareReceivingData=${urlSafeData}`)
    } catch (e) {
      // Handle ShareExtension error
    }
  }

You might also want to share storage between your host app and share extension to do things like check login status and only launch the host app if someone is logged out, etc. Everything is now a possibility.

**Comment by BigPun86** [comment]
@isaachinman thanks man. I actually got it working exactly like this using one PR where they have implemented openURL natively into the share-extensions AppDelegate. Now this works all great when the Main App is opened or in background etc. But it does not work when the app is completely closed. Maybe react-native-swiss-knife will solve this, i will test it tomorrow

**Comment by mtzfactory** [comment]
Hi @BigPun86,

Have you tried yet?

**Comment by BigPun86** [comment]
@mtzfactory yes, but unfortunately i couldn't get it working! Actually i have fixed it in a separate fork => https://github.com/BigPun86/react-native-share-extension

There will be a much cleaner implementation. I will push it the next days, hopefully until the weekend.

**Comment by ekeuus** [comment]
@BigPun86 Were you able to solve this with the "clean" implementation?

**Comment by isaachinman** [comment]
@frenberg Feel free to check out #67, and other places where we've discussed this. Many of us have released apps via Apple containing this approach. Moreover, large apps like Pinterest take this approach, as @birkir points out.

You are correct in that it may be against Apple policy on paper, but in practice we've had no issues in reviews. The risk associated with a failed review is something to discuss with your team/colleagues.

**Comment by frenberg** [comment]
@isaachinman thank you for that info, that was new to me. Very interesting!

Tried to have a look at how Pintrest do this but their share extension just flickers and crashes on my iphone xs :)

**Comment by theone3nu** [comment]
@isaachinman I have tried this 

```
> async componentDidMount() {
  try {
    const data = await ShareExtension.data()
    const urlSafeData = JSON.stringify({ data })
    await ShareExtension.openURL(`myapp://ShareReceive?shareReceivingData=${urlSafeData}`)
  } catch (e) {
    // Handle ShareExtension error
  }
}
```
But my react native app is not opening. Can you please help?

**Comment by isaachinman** [comment]
Are you on Android? I believe you need to take a different approach vs iOS:

```jsx
await ShareExtension.openURL(url) // iOS
await Linking.openURL(url) // Android
```

**Comment by theone3nu** [comment]
@isaachinman Actually code is not updated in npm repo. But it is updated in github. I directly downloaded the code from github and used that library then it fixed the issue. Thanks for the library.

**Comment by isaachinman** [comment]
@AndrewHenderson Do we have any release targets currently? This project hasn't been deployed on npm in a year.

**Comment by daviswhitehead** [comment]
@AndrewHenderson @isaachinman I just want to bump that I was able to resolve this as @theone3nu  did by downloading the repo directly from github. It would be really great to get another up to date npm deploy... Not sure who's responsible for that, maybe @alinz ?

**Comment by isaachinman** [comment]
@daviswhitehead I've got `npm` access. You're saying you've been using the latest `master` as a dep, right?

**Comment by daviswhitehead** [comment]
Yep that's correct, I just now downloaded `master` and replaced the existing `node_modules/react-native-share-extension` folder with its files and it worked perfectly.

**Comment by isaachinman** [comment]
@AndrewHenderson Unless you object to it, I am going to release the current master as a major (v2.0.0) imminently. I'm not sure if there are breaking changes in the diff so this seems safest to me.

**Comment by isaachinman** [comment]
I've just released v2.0.0.

**Comment by AndrewHenderson** [comment]
@isaachinman Apologies for the delayed response. I think it was good idea to release the current master branch. There were bugs in the previous npm version. 👍

There wasn't anything major in this release. I believe it was mostly bug fixes.

The features I believe still need to be developed are listed here: https://github.com/alinz/react-native-share-extension/issues/123#issuecomment-437470240.

**Comment by SirCameron** [comment]
I'm having this issue as well and I cannot find the setting "Require Only App-Extension-Safe API" in the extension build settings.. Xcode 10.3

**Comment by colaskirschoff** [comment]
@SirCameron have you tried using the search bar in build settings ?

<img width="966" alt="image" src="https://user-images.githubusercontent.com/9667205/63935666-42534200-ca5e-11e9-8048-2c6b9579ce41.png">
